reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -1717,7 +1718,7 @@ public T block(Duration timeout) {
 	 *
 	 * @return a new {@link Mono}
 	 */
-	public final Mono<T> doOnSubscribe(Consumer<? super Subscription> onSubscribe) {
+	public final Mono<@NonNull T> doOnSubscribe(Consumer<? super @NonNull Subscription> onSubscribe) {
 		Objects.requireNonNull(onSubscribe, "onSubscribe");
 		return doOnSignal(this, onSubscribe, null, null,  null, null, null);
 	}
```

Comment:
the `doOnSuccessOrError` below should have its `BiConsumer` generic types marked as `Nullable` both, like `doAfterSuccessOrError`

Reply:
Wrong line I guess, I have done the update on the `BiConsumer` mentioned.

Reply:
yes, couldn't comment on the line as it was not part of the diff, 👍